### PR TITLE
Update, clean, and reorganize the Readme files.

### PR DESCRIPTION
**Pull request by sjrd**

### CODE REVIEW

**Comment by sjrd** [comment]
Review by @gzm0, but no rush at all.

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+## Pull Request Requirements
+
+In order for a Pull Request to be considered, it has to meet these requirements:
+
+1.  Live up to the current code standard:
+    - Not violate [DRY](http://programmer.97things.oreilly.com/wiki/index.php/Don%27t_Repeat_Yourself).
+    - [Boy Scout Rule](http://programmer.97things.oreilly.com/wiki/index.php/The_Boy_Scout_Rule) should be applied.
+2.  Be accompanied by appropriate tests.
+
+If *all* of these requirements are not met then the code should **not** be
```
Comment (gzm0) [inline CONTRIBUTING.md]: If not *all* of these requirements... (this actually does not mean what you want it to)

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+    For more info, see the article:
+    [Git Workflow](http://sandofsky.com/blog/git-workflow.html).
+    Additionally, every commit should be able to be used in isolation--that is,
+    each commit must build and pass all tests.
+2.  The first line should be a descriptive sentence about what the commit is doing.
+    It should be possible to fully understand what the commit does by just
+    reading this single line.
+    It is **not ok** to only list the ticket number, type "minor fix" or similar.
+    If the commit has a corresponding ticket, include a reference to the ticket
+    number, with the format "Fix #xxx: Title of the ticket.", as the first line,
```
Comment (gzm0) [inline CONTRIBUTING.md]: Don't agree here. It should be: "Fix #xxx: What does this change". Ticket titles are very often: xyz fails. That's not a very good commit message.

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+    assuming that it aptly and concisely summarizes the commit in a single line.
```
Comment (gzm0) [inline CONTRIBUTING.md]: Oh... just saw that... maybe you could put my example here as a situation where you should deviate from the above.

**Comment by gzm0** [comment]
That's all

File: CONTRIBUTING.md (hunk 2 above)
Comment (sjrd) [inline CONTRIBUTING.md]: That's what we've been doing most of the time, though. But true, it's not always the best message. I'll elaborate a bit.

Comment (gzm0) [inline CONTRIBUTING.md]: > That's what we've been doing most of the time, though. 

I know. It always bugged me :P

**Comment by sjrd** [comment]
Updated.

**Comment by gzm0** [comment]
LGTM

**Pull request merged**